azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -70,22 +70,44 @@ private void restartExecutableFlow(final ExecutableFlow exFlow) {
     // Update the flow options so that the flow will be not retried again by Azkaban
 
     // inherent the retry time counters
-    executableFlow.setUserDefinedRetryCount(exFlow.getUserDefinedRetryCount());
-    executableFlow.setSystemDefinedRetryCount(exFlow.getSystemDefinedRetryCount());
+    retryExFlow.setUserDefinedRetryCount(originalExFlow.getUserDefinedRetryCount());
+    retryExFlow.setSystemDefinedRetryCount(originalExFlow.getSystemDefinedRetryCount());
+
+    if (originalExFlow.getFlowRetryRootExecutionID() > 0) {
+      retryExFlow.setFlowRetryRootExecutionID(originalExFlow.getFlowRetryRootExecutionID());
+    } else {
+      retryExFlow.setFlowRetryRootExecutionID(originalExFlow.getExecutionId());
+    }
+    retryExFlow.setFlowRetryParentExecutionID(originalExFlow.getExecutionId());
 
     // If a retried flow A gets retried again with a new execution id flow B, the original flow
     // execution id of flow B should be the same as flow A's original flow execution id.
     if (options.getOriginalFlowExecutionIdBeforeRetry() == null) {
```

Comment:
I don't really get what this OriginalFlowExecutionBeforeRetry means. And can you remind me why this has to be put into executionOptions?